### PR TITLE
BitMask documentation and some extraneous other doc changes

### PR DESCRIPTION
**Pull request by kbwestfall**
This adds documentation for the bitmasks, as well as reorganizes a few of the documentation files (most annoyingly `links.rst`).  I also added a script that can be used to generate the help messages for the scripts so that they can be dynamically updated in the documentation.  I used `pypeit_show_2dspec` as an example and showed that it works.

Note this is being directed into the branch for current PR #1033 because I mistakenly branched off it to create this branch.  This keeps the changes separate, but we should essentially merge this into PR #1033 and then merge that to `develop`.

### CODE REVIEW

File: doc/help/pypeit_show_2dspec.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.. code-block:: console
```
Comment (profxj) [inline doc/help/pypeit_show_2dspec.rst:1]: Are these being generated on-the-fly?  Just checking.

File: doc/include/bitmask_usage.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
+
+Bitmasks allow you to define a set of bit values signified by strings,
+and then toggle and interpret bits held by a `numpy.ndarray`_.  For
+example, say you're processing an image and you want to set up a set of
+bits that indicate that the pixel is part of a bad-pixel mask, has a
+cosmic ray, or is saturated.  You can define the following::
+
+    from collections import OrderedDict
```
Comment (profxj) [inline doc/include/bitmask_usage.rst:8]: You might drop the `OrderedDict` part.  We are fully on Python >= 3.7

File: doc/include/imagebitmask_table.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
+==========  =========  ===================================================
```
Comment (profxj) [inline doc/include/imagebitmask_table.rst:1]: If it is easy to do, maybe add another column which is `2**BIT_VALUE`

File: doc/out_masks.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    - (Recommended) Instantiate the bitmask directly:
+
+      .. code-block:: python
+
+        from pypeit.images.imagebitmask import ImageBitMask
+        bm = ImageBitMask()
+
+    - Specifically for the ``spec2d*`` files, the names of the bits
+      and their order is save to the header. You can instantiate a
```
Comment (profxj) [inline doc/out_masks.rst:42]: saved

File: doc/scripts/build_bitmask_rst.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+def write_bitmask_table(obj, path):
+    bm = obj()
+    ofile = os.path.join(path, '{0}_table.rst'.format(obj.__name__.lower()))
+
+    data_table = numpy.empty((bm.nbits+1, 3), dtype=object)
+    data_table[0,:] = ['Bit Name', 'Bit Value', 'Description']
+    for i,k in enumerate(bm.bits.keys()):
+        data_table[i+1,0] = k
+        data_table[i+1,1] = to_string(bm.bits[k])
```
Comment (profxj) [inline doc/scripts/build_bitmask_rst.py:24]: Consider adding a `2**bm.bits[k]` column for those that
don't quite get the concept of a bit.  :)

File: doc/scripts/write_script_help.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+"""
+Dynamically build the rst documentation with the script help text.
```
Comment (profxj) [inline doc/scripts/write_script_help.py:2]: Nice!  Feel free to add in any other script you have the
endurance to handle.

**Review by profxj — APPROVED** [review]
Just a few minor suggestions on my end.

Thanks for this!

File: doc/help/pypeit_show_2dspec.rst (hunk 1 above)
Comment (kbwestfall) [reply doc/help/pypeit_show_2dspec.rst:1]: Yep.

File: doc/include/bitmask_usage.rst (hunk 2 above)
Comment (kbwestfall) [reply doc/include/bitmask_usage.rst:8]: Dropped and added some explanation.

File: doc/include/imagebitmask_table.rst (hunk 3 above)
Comment (kbwestfall) [reply doc/include/imagebitmask_table.rst:1]: Done.

File: doc/out_masks.rst (hunk 4 above)
Comment (kbwestfall) [reply doc/out_masks.rst:42]: Good catch!  Fixed.

File: doc/scripts/build_bitmask_rst.py (hunk 5 above)
Comment (kbwestfall) [reply doc/scripts/build_bitmask_rst.py:24]: Done.

File: doc/scripts/write_script_help.py (hunk 6 above)
Comment (kbwestfall) [reply doc/scripts/write_script_help.py:2]: Incoming...

**Review by badpandabear — APPROVED** [review]

**Comment by kbwestfall** [comment]
I put in the time to produce these `help/*.rst` files for all scripts in pypeit.  This means that I've touched *all* of the scripts.   Tests pass:

```========== 246 passed in 984.73s (0:16:24) ==========```

This exercise shows that the following scripts are not discussed in the docs:

```
pypeit_chk_alignments
pypeit_chk_tilts
pypeit_coadd_datacube
pypeit_lowrdx_skyspec
pypeit_qa_html
pypeit_ql_keck_mosfire
pypeit_show_arxiv
pypeit_show_wvcalib
pypeit_skysub_regions
```

Let me know if you want me to run the devsuite before merging.

**Comment by profxj** [comment]
Merge when ready

**Pull request merged**